### ISSUE TITLE
Mockito arguments matcher

### ISSUE THREAD

**Issue by nkpodolyan** [flutter/flutter#20253]
I am experiencing an issue with flutter widget test, using mockito`s argument matcher.
 
I am using flutter beta channel (Flutter 0.5.1 • channel beta) here is my  pubspec.yaml:
```
..............................
dependencies:
  flutter:
    sdk: flutter
  cupertino_icons: ^0.1.2
  flutter_redux: ^0.5.2
  redux: ^3.0.0
  rxdart: ^0.18.1
  meta: ^1.1.5
  quiver: ^0.29.0+1
  built_value: <=5.5.0
  built_collection: ^3.1.1
  redux_epics: ^0.9.0
  flutter_logger: 0.0.2
  random_words: ^1.0.1

dependency_overrides:
  redux:
    git: https://github.com/johnpryan/redux.dart
  rxdart:
    git: https://github.com/ReactiveX/rxdart.git

dev_dependencies:
  build_runner: ^0.8.9
  built_value_generator: <=5.5.0
  mockito: ^2.2.3
  flutter_test:
      sdk: flutter

flutter:
  assets:
    - res/image/
  uses-material-design: true

```

The code of class I am mocking using mockito:
```
class ViewModel {
  final void Function(String title) onTextEntered;
  .................
}
```

and the test:
```
class ViewModelMock extends Mock implements ViewModel {}

void main() {
  group("Description", () {
    ViewModelMock mockModel;
    Widget widgetUnderTes;

    setUpAll(() async {
      mockModel = ViewModelMock();
      widgetUnderTes = ListWidget(mockModel);
    });

    testWidgets('Test description', (WidgetTester tester) async {
      // When
      await tester.pumpWidget(MaterialApp(
        home: widgetUnderTes,
      ));

      await tester.tap(find.byType(FloatingActionButton));

     // exception is thrown here:
      verify(mockModel.onTextEntered(any));
    });
  });
}
```
and the exception itself:

```
══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞════════════════════════════════════════════════════
The following assertion was thrown running a test:
type 'ArgMatcher' is not a subtype of type 'String'

Either the assertion indicates an error in the framework itself, or we should provide substantially
more information in this error message to help you determine and fix the underlying cause.
In either case, please report this assertion by filing a bug on GitHub:
  https://github.com/flutter/flutter/issues/new

When the exception was thrown, this was the stack:
#0      main.<anonymous closure>.<anonymous closure> (file:///<path_to_project>/test/widget/widget_test.dart:36:41)
<asynchronous suspension>
#1      testWidgets.<anonymous closure>.<anonymous closure> (package:flutter_test/src/widget_tester.dart:65:23)
#2      TestWidgetsFlutterBinding._runTestBody (package:flutter_test/src/binding.dart:502:19)
<asynchronous suspension>
#5      TestWidgetsFlutterBinding._runTest (package:flutter_test/src/binding.dart:486:14)
#6      AutomatedTestWidgetsFlutterBinding.runTest.<anonymous closure> (package:flutter_test/src/binding.dart:765:24)
#12     AutomatedTestWidgetsFlutterBinding.runTest (package:flutter_test/src/binding.dart:763:16)
#13     testWidgets.<anonymous closure> (package:flutter_test/src/widget_tester.dart:64:22)
#14     Declarer.test.<anonymous closure>.<anonymous closure>.<anonymous closure> (package:test/src/backend/declarer.dart:161:27)
<asynchronous suspension>
#15     Invoker.waitForOutstandingCallbacks.<anonymous closure> (package:test/src/backend/invoker.dart:249:15)
<asynchronous suspension>
#19     Invoker.waitForOutstandingCallbacks (package:test/src/backend/invoker.dart:246:5)
#20     Declarer.test.<anonymous closure>.<anonymous closure> (package:test/src/backend/declarer.dart:159:33)
#24     Declarer.test.<anonymous closure> (package:test/src/backend/declarer.dart:158:13)
<asynchronous suspension>
#25     Invoker._onRun.<anonymous closure>.<anonymous closure>.<anonymous closure>.<anonymous closure> (package:test/src/backend/invoker.dart:403:25)
<asynchronous suspension>
#39     _Timer._runTimers (dart:isolate/runtime/libtimer_impl.dart:382:19)
#40     _Timer._handleMessage (dart:isolate/runtime/libtimer_impl.dart:416:5)
#41     _RawReceivePortImpl._handleMessage (dart:isolate/runtime/libisolate_patch.dart:165:12)
(elided 26 frames from class _FakeAsync, package dart:async, and package stack_trace)
```

I am using mockito: ^2.2.3 because it is compatible with a flutter_test framework. I figured out that mockito: ^3.0.0 has fix for an issue I am experiencing, but it is not compatible with flutter_test framework (Because mockito >=3.0.0-beta+3 depends on test ^1.2.0 and every version of flutter_test from sdk depends on test 0.12.37, mockito >=3.0.0-beta+3 is incompatible with flutter_test from sdk).

I also checked that mockito: ^3.0.0 is incompatible with all versions of flutter (dev & master).

Please advise how to overcome "type 'ArgMatcher' is not a subtype of type 'String'" issue?

**Comment by bizz84** [comment]
I bumped into a similar issue myself.

In my case I was able to solve it by changing the expectation to use `typed(argThat(isNotNull)))`. In your case `typed(any)` may work:

```
verify(mockModel.onTextEntered(typed(argThat(isNotNull))));
```

The documentation for `typed` is not easy to find, but I found something here:

https://github.com/dart-lang/mockito/blob/master/upgrading-to-mockito-3.md

This error is super annoying:

> Because mockito >=3.0.0-beta+3 depends on test ^1.2.0 and every version of flutter_test from sdk depends on test 0.12.37, mockito >=3.0.0-beta+3 is incompatible with flutter_test from sdk.

I think this really needs to be sorted as I believe no-one can use mockito 3 on any Flutter branch.
